R/teal_slice-store.R

```diff
@@ -51,6 +53,22 @@ slices_restore <- function(file) {
   checkmate::assert_file_exists(file, access = "r", extension = "json")
 
   tss_json <- jsonlite::fromJSON(file, simplifyDataFrame = FALSE)
+  tss_json$slices <-
+    lapply(tss_json$slices, function(slice) {
+      if (!is.null(slice$selected)) {
+        slice$selected <-
```

Comment:
How about the `choices`? I think we may want to restore it as it can take `Date` or `POSIXct`

The test below fails

```R
test_that("teal_slice store/restore supports saving `Date` timestamps in choices", {
  slices_path <- withr::local_file("slices.json")

  time_stamps <- Sys.Date() + c(-10 * 600, -30, 0)

  tss <- teal_slices(
    teal_slice(
      dataname = "ADSL",
      varname = "COUNTRY",
      selected = sample(time_stamps, 2),
      choices = time_stamps,
      fixed = TRUE
    )
  )

  # Store the teal_slices object to a file
  slices_store(tss, slices_path)
  tss_restored <- slices_restore(slices_path)

  tss_restored_list <- shiny::isolate(shiny::reactiveValuesToList(tss_restored[[1]]))
  expect_s3_class(tss_restored_list$choices, "Date")

  expect_identical_slice(tss[[1]], tss_restored[[1]])
})
```

Reply:
Thanks for the input @averissimo - valid point

Reply:
Thanks, just extended in here d7aebc9cac60ac4979719ef40888fdbd0d2d4b13